workflows/diagnostics/tests/offline/test_compute_diagnostics.py

```diff
@@ -0,0 +1,161 @@
+import numpy as np
+import pytest
+import xarray as xr
+
+from fv3net.diagnostics.offline.compute_diagnostics import (
+    weighted_average,
+    _snap_mask_to_type,
+    _snap_net_precipitation_to_type,
+    _conditional_average,
+    merge_diagnostics,
+)
+
+
+da = xr.DataArray(np.arange(1.0, 5.0), dims=["z"])
+da_nans = xr.DataArray(np.full((4,), np.nan), dims=["z"])
+ds = xr.Dataset({"a": da})
+weights = xr.DataArray([0.5, 0.5, 1, 1], dims=["z"])
+weights_nans = xr.DataArray(np.full((4,), np.nan), dims=["z"])
+
+
```

Comment:
Tests copied from `diagnostics_utils`, no new code. The tests from `test_compute_metrics` have also been moved here.